rasa/model_training.py

```diff
@@ -218,11 +218,10 @@ async def _train_async_internal(
     new_fingerprint = await model.model_fingerprint(file_importer)
     old_model = model.get_latest_model(output_path)
 
-    fingerprint_comparison = model.should_retrain(
-        new_fingerprint, old_model, train_path, force_training=force_training
-    )
-
     if dry_run:
+        fingerprint_comparison = model.should_retrain(
```

Comment:
Would it be possible to leave this above `if dry_run` but use the result later in the method instead of re-running?

Reply:
The two calls to model.should_retrain() are passing different numbers of arguments and I have not investigated further into the subroutine's code to understand how to combine them into a single call.

Reply:
Ok, this should be fine for now then, as we will be refactoring this area of code soon anyway.